src/read.rs

```diff
@@ -31,27 +31,24 @@ mod ffi {
     pub const S_IFREG: u32 = 0o0100000;
 }
 
-/// Wrapper for reading the contents of a ZIP file.
+/// ZIP archive reader
 ///
 /// ```no_run
-/// use std::io::prelude::*;
-/// fn main() -> zip::result::ZipResult<()> {
+/// # use std::io::prelude::*;
+/// # fn main() -> zip::result::ZipResult<()> {
+/// # let buf: &[u8] = &[0u8; 128];
+/// # let mut reader = std::io::Cursor::new(buf);
+/// # let mut stdout = std::io::stdout();
+/// # let mut stdout = stdout.lock();
+/// let mut zip = zip::ZipArchive::new(reader)?;
```

Comment:
I personally don't like examples where the user can't easily copy/paste the code and have it run. It can be confusing to new-comers what the `reader` and `buf` actually are.

Reply:
This *is* a no_run test - the uncommented code would always crash. We'd need to provide valid zip data to complete the example, which seems like it'd be distracting

Reply:
I'd add to that that `buf` isn't visible in the test. All the users need to recognize is that `reader` is referring to *any* `impl Read`, which I think is fairly reasonable. We could change this to a function that takes an `impl Read` itself, but that seems unnecessary to me

Reply:
I'd add to that that `buf` isn't visible in the test. All the users need to recognize is that `reader` is referring to *any* `impl Read`, which I think is fairly reasonable. We could change this to a function that takes an `impl Read` itself, but that seems unnecessary to me

Reply:
Again, I don't like examples that aren't directly copy/pastable, but if you feel strongly that this is an improvement, we can leave it.

Reply:
I'm not clear what you consider better. Would you like the function signature to be uncommented?

Reply:
I'm not clear what you consider better. Would you like the function signature to be uncommented?

Reply:
I would prefer that yes. Or provide a concrete `Read` type as a local variable.